src/v6/htype.rs

```diff
@@ -0,0 +1,202 @@
+    FrameRelay,
+    /// 16,19,21 ATM
+    ATM,
+    /// 17 HDLC
+    HDLC,
+    /// 18 FibreChannel
+    FibreChannel,
+    /// 20 SerialLine
+    SerialLine,
+    /// 22 Mil STD
+    MilStd188220,
+    /// 23 Metricom
+    Metricom,
+    /// 24 IEEE1394.1995
+    IEEE13941995,
+    /// 25 MAPOS
+    MAPOS,
+    /// 26 Twinaxial
+    Twinaxial,
+    /// 27 EUI64
+    EUI64,
+    /// 28 HIPARP
+    HIPARP,
+    /// 29 IP and ARP over ISO 7816-3
+    IPandARPoverISO78163,
+    /// 30 ARPSec
+    ARPSec,
+    /// 31 IPsec tunnel
+    IPsecTunnel,
+    /// 32 Infiniband
+    Infiniband,
+    /// 33 TIA-102 Project 25 Common Air Interface (CAI)
+    CAI,
+    /// 34 WeigandInt
+    WiegandInt,
+    /// 35 PureIP
+    PureIP,
+    /// 36 HW_EXP1
+    HWExp1,
+    /// 37 HFI
+    HFI,
+    /// 38 Unified BUS(UB),
+    UB,
+    /// 256 HW_EXP2
+    HWExp2,
+    /// 257 AEthernet
+    AEthernet,
+    /// 65535 Reserved
+    Reserved,
+    /// Unknown or not yet implemented htype
+    Unknown(u16),
+}
+impl From<u16> for HType {
+    fn from(n: u16) -> Self {
+        use HType::*;
+        match n {
+            1 => Eth,
+            2 => ExperimentalEth,
+            3 => AmRadioAX25,
+            4 => ProteonTokenRing,
+            5 => Chaos,
+            6 => IEEE802,
+            7 => ARCNET,
+            8 => Hyperchannel,
+            9 => Lanstar,
+            10 => AutonetShortAddr,
+            11 => LocalTalk,
+            12 => LocalNet,
+            13 => Ultralink,
+            14 => SMDS,
+            15 => FrameRelay,
+            16 | 19 | 21 => ATM,
+            17 => HDLC,
+            18 => FibreChannel,
+            20 => SerialLine,
+            22 => MilStd188220,
+            23 => Metricom,
+            24 => IEEE13941995,
+            25 => MAPOS,
+            26 => Twinaxial,
+            27 => EUI64,
+            28 => HIPARP,
+            29 => IPandARPoverISO78163,
+            30 => ARPSec,
+            31 => IPsecTunnel,
+            32 => Infiniband,
+            33 => CAI,
+            34 => WiegandInt,
+            35 => PureIP,
+            36 => HWExp1,
+            37 => HFI,
+            38 => UB,
+            256 => HWExp2,
+            257 => AEthernet,
+            65535 => Reserved,
+            n => Unknown(n),
+        }
+    }
+}
+
+impl From<HType> for u16 {
+    fn from(n: HType) -> Self {
+        use HType as H;
+        match n {
+            H::Eth => 1,
+            H::ExperimentalEth => 2,
+            H::AmRadioAX25 => 3,
+            H::ProteonTokenRing => 4,
+            H::Chaos => 5,
+            H::IEEE802 => 6,
+            H::ARCNET => 7,
+            H::Hyperchannel => 8,
+            H::Lanstar => 9,
+            H::AutonetShortAddr => 10,
+            H::LocalTalk => 11,
+            H::LocalNet => 12,
+            H::Ultralink => 13,
+            H::SMDS => 14,
+            H::FrameRelay => 15,
+            H::ATM => 16, // !!! 16,19,21 are all possible. 19 is chosen according to RFC2225
```

Comment:
We could just remove this variant, and people could use `Unknown` for these values. I think all o these becoming ATM could be surprising.

Reply:
Ok